scripts/clean-up-localizations.js

```diff
@@ -0,0 +1,65 @@
+const fs = require('fs');
+const glob = require('glob');
+
+const localizationFile_en = 'src/assets/i18n/en_US.json';
+const localizationFile_fr = 'src/assets/i18n/fr_FR.json';
+const localizationFile_de = 'src/assets/i18n/de_DE.json';
+
+// regular expressions for patterns of not explicit used localization keys (dynamic created keys, error keys from rest calls)
+// new patterns have to be added here
+const patterns = ['account.login..*.message', '.*.?error..*'];
+const regEx = new RegExp(patterns.join('|'), 'i');
+
+var filesToBeSearched = [];
+var localizationsFound = {};
+var localizationsFoundOrdered = {};
+
+// store localizations from default localization file in an object
+var localizations = JSON.parse(fs.readFileSync(localizationFile_en, 'utf8'));
+
+var localizations_fr = JSON.parse(fs.readFileSync(localizationFile_fr, 'utf8'));
+var localizations_de = JSON.parse(fs.readFileSync(localizationFile_de, 'utf8'));
+
+// go through directory recursively and find files to be searched
+glob.sync('{src,projects}/**/!(*.spec).{ts,html}').forEach(filePath => {
+  filesToBeSearched.push(filePath);
+});
+
+// add not explicit used localization keys
+Object.keys(localizations)
+  .filter(v => v.match(regEx))
```

Comment:
String.[match](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/String/match) returns an array with matches. In this case I think we just want to have a boolean, if the key matches the regular expression. RegExp.[test](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test) is more suitable for this.